src/_guides/language/concurrency/index.md

```diff
@@ -0,0 +1,403 @@
+
+Even single-isolate programs can execute smoothly
+by using async-await to wait for asynchronous operations to complete
+before continuing to the next line of code.
+A well-behaved app starts quickly,
+getting to the event loop as soon as possible.
+The app then responds to each queued event promptly,
+using asynchronous operations as necessary.
+
+As the following figure shows,
+every isolate starts by running some Dart code,
+such as the `main()` function.
+This Dart code might register some event listeners —
+to respond to user input or file I/O, for example.
+When the isolate's initial function returns,
+the isolate stays around if it needs to handle events.
+After handling the events, the isolate exits.
+
+![A more general figure showing that any isolate runs some code, optionally responds to events, and then exits](/guides/language/concurrency/images/basics-isolate.png)
+
+In a client app, the main isolate’s event queue might contain
+repaint requests and notifications of tap and other UI events, like this:
+***[PENDING: describe the figure in more detail.]***
+
+![PENDING: alt text here](/guides/language/concurrency/images/event-loop.png)
+
+The event handling happens on the main isolate after `main()` exits:
+***[PENDING: describe the figure in more detail.]***
+
+![PENDING: alt text here](/guides/language/concurrency/images/event-handling.png)
+
+If a synchronous operation takes too much processing time,
+the app can become unresponsive.
+In the following figure, the tap-handling code takes too long,
+so subsequent events are handled too late:
+***[PENDING: describe the figure in more detail.]***
+
+![PENDING: alt text here](/guides/language/concurrency/images/event-jank.png)
+
+In client apps, the result of a too-lengthy synchronous operation is often
+[janky (non-smooth) UI animation][jank].
+Worse, the UI might become completely unresponsive.
+
+[jank]: https://docs.flutter.dev/perf/rendering
+
+
+### Using a single-task background worker
+
+If your app’s UI becomes unresponsive due to a time-consuming computation —
+[parsing a large JSON file][json], for example —
+consider offloading that computation to a background worker.
+A common case, which is supported by the [Flutter `compute()` function][],
+is spawning an isolate that performs a computation and then exits:
+***[PENDING: describe the figure in a little more detail.]***
+
+[json]: https://docs.flutter.dev/cookbook/networking/background-parsing)
+ [Flutter `compute()` function]: https://docs.flutter.dev/cookbook/networking/background-parsing#4-move-this-work-to-a-separate-isolate
+
+![PENDING: alt text here](/guides/language/concurrency/images/isolate-bg-worker.png)
+
+A background worker can perform I/O
+(reading and writing files, for example), set timers, and more.
+It has its own memory (including its own copy of code) and
+doesn’t share any state with the main isolate.
+
```

Comment:
Could we add sample code here for this "single-task background worker" use case ?

I find it confusing to read the sample code further down on the page as it follows a diagram that shows a worker isolate that is not single-task. I think the docs would be easier to read with a structure like this:

```
### Using a single-task background worker

Intro text.

Diagram.

Sample-code using Isolate.exit().

### Using a multi-task background worker

Intro text.

Diagram.

Sample-code using Isolate.send()

```